src/Storages/Statistics/Statistics.cpp

```diff
@@ -124,10 +97,16 @@ Float64 ColumnStatistics::estimateEqual(const Field & val) const
     if (stats.contains(StatisticsType::CountMinSketch))
         return stats.at(StatisticsType::CountMinSketch)->estimateEqual(val);
 #endif
-    if (!float_val.has_value() && (float_val < - ConditionSelectivityEstimator::threshold || float_val > ConditionSelectivityEstimator::threshold))
-        return rows * ConditionSelectivityEstimator::default_normal_cond_factor;
-    else
-        return rows * ConditionSelectivityEstimator::default_good_cond_factor;
+    if (stats.contains(StatisticsType::Uniq))
+    {
+        auto cardinality = stats.at(StatisticsType::Uniq)->estimateCardinality();
```

Comment:
Let's write the type out instead of `auto` if the right side does not make it immediately clear.